api/services/MachineService.js

```diff
@@ -322,7 +330,23 @@ function sessionOpen(user) {
  * @return {Promise}
  */
 function sessionEnded(user) {
-  return increaseUsersMachineEndDate(user);
+  var ret = increaseUsersMachineEndDate(user);
+  if (_driver.name() === 'aws') {
```

Comment:
Don't depend of the driver name. You may want to add a method `isUserCreditSupported` or something.